flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/Dispatcher.java

```diff
@@ -603,7 +616,7 @@ private JobManagerRunner startJobManagerRunner(JobManagerRunner jobManagerRunner
 
 	@Override
 	public CompletableFuture<Acknowledge> shutDownCluster() {
-		closeAsync();
+		shutDownFuture.complete(ApplicationStatus.SUCCEEDED);
```

Comment:
So we complete `shutDownFuture` here and cluster entrypoint takes care of the close of the whole cluster?

Another concern is `ApplicationStatus.SUCCEEDED` v.s. `ApplicationStatus.CANCEL` since, IIRC, shutDownCluster is called via REST so it means user shutdown the cluster intendedly.

Reply:
And follow the previous pull request we actually complete with `ApplicationStatus.UNKNOWN` in this case.

Reply:
The reason why we complete the shut down future with `SUCCEEDED` is to imitate the job cluster behaviour when submitting multi staged jobs (which under the hood spawns a session cluster). It is not perfect but a workaround to not confuse people when they see that the Yarn application has been canceled if they submitted a job in per-job mode. I think it is ok for the moment and can be changed to `CANCELED` once we have proper per-job mode support.

Reply:
Agree revisit this part when we have proper per-job mode support. Noted.